eth/api.go

```diff
@@ -671,3 +671,30 @@ func (api *ScrollAPI) GetLatestRelayedQueueIndex(ctx context.Context) (queueInde
 	lastIncluded := *queueIndex - 1
 	return &lastIncluded, nil
 }
+
+// rpcMarshalBlock uses the generalized output filler, then adds the total difficulty field, which requires
+// a `ScrollAPI`.
+func (api *ScrollAPI) rpcMarshalBlock(ctx context.Context, b *types.Block, inclTx bool, fullTx bool) (map[string]interface{}, error) {
+	fields, err := ethapi.RPCMarshalBlock(b, inclTx, fullTx, api.eth.APIBackend.ChainConfig())
+	if err != nil {
+		return nil, err
+	}
+	if inclTx {
+		fields["totalDifficulty"] = (*hexutil.Big)(api.eth.APIBackend.GetTd(ctx, b.Hash()))
+	}
+	rc := rawdb.ReadBlockRowConsumption(api.eth.ChainDb(), b.Hash())
+	if rc != nil {
+		fields["rowConsumption"] = hexutil.Uint64(rc.Rows)
+	}
```

Comment:
If we add `else fields["rowConsumption"] = nil`, would that be serialized into JSON? If yes, that might be better, at least it's more explicit when this data is missing.

Reply:
Yes, added in new commit